cmd/osbuild-worker/main.go

```diff
@@ -148,32 +149,47 @@ func main() {
 	var client *worker.Client
 	if unix {
 		client = worker.NewClientUnix(address)
-	} else if config.Authentication != nil && config.Authentication.OfflineTokenPath != "" {
-		t, err := ioutil.ReadFile(config.Authentication.OfflineTokenPath)
-		if err != nil {
-			log.Fatalf("Could not read offline token: %v", err)
-		}
-		token := strings.TrimSpace(string(t))
+	} else {
+		var token *string
+		var oAuthURL *string
+		if config.Authentication != nil && config.Authentication.OfflineTokenPath != "" {
+			t, err := ioutil.ReadFile(config.Authentication.OfflineTokenPath)
+			if err != nil {
+				log.Fatalf("Could not read offline token: %v", err)
+			}
+			t2 := strings.TrimSpace(string(t))
+			token = &t2
 
-		if config.Authentication.OAuthURL == "" {
-			log.Fatal("OAuth URL should be specified together with the offline token")
+			if config.Authentication.OAuthURL == "" {
+				log.Fatal("OAuth URL should be specified together with the offline token")
+			}
+			oAuthURL = &config.Authentication.OAuthURL
+
+			if strings.HasPrefix(address, "http") {
+				out, err := exec.Command("systemd-escape", "-u", address).Output()
```

Comment:
Isn't there a way to escape this without using `systemd`?

(If not, we might need to tweak the spec file to require this executable.)

Reply:
Ah good point, I'll look into it

Reply:
Hmm, systemd-escape is shipped as part of systemd; so we should already be covered since we already require systemd.
![image](https://user-images.githubusercontent.com/11140201/131529190-df1cd0ec-465d-4c29-b361-23554c7b793f.png)